container-registry/container-analysis/src/main/java/com/example/containeranalysis/Samples.java

```diff
@@ -43,307 +52,288 @@
-      occBuilder.setResourceUrl(imageUrl);
-      VulnerabilityDetails vd = VulnerabilityType.VulnerabilityDetails.newBuilder().build();
-      occBuilder.setVulnerabilityDetails(vd);
-      Occurrence newOcc = occBuilder.build();
-      return client.createOccurrence(projectName, newOcc);
-    }
+  public static Occurrence createOccurrence(GrafeasV1Beta1Client client, String imageUri, 
+      String parentNoteId,String projectId) {
+    final String parentNoteName = client.formatNoteName(projectId, parentNoteId);
+    final String projectName = client.formatProjectName(projectId);
+
+    Occurrence.Builder occBuilder = Occurrence.newBuilder();
+    occBuilder.setNoteName(parentNoteName);
+    Details.Builder detailsBuilder = Details.newBuilder();
+    // Details about the vulnerability instance can be added here
+    occBuilder.setVulnerability(detailsBuilder);
+    // Attach the occurrence to the associated image uri
+    Resource.Builder resourceBuilder = Resource.newBuilder();
+    resourceBuilder.setUri(imageUri);
+    occBuilder.setResource(resourceBuilder);
+    Occurrence newOcc = occBuilder.build();
+    return client.createOccurrence(projectName, newOcc);
   }
   // [END create_occurrence]
 
   // [START update_note]
   /**
-   * Makes an update to an existing note
-   *
+   * Pushes an update to a Note that already exists on the server
+   * @param client The Grafeas client used to perform the API requests.
    * @param updated a Note object representing the desired updates to push
    * @param noteId the identifier of the existing note
-   * @param projectId the GCP project the occurrence will be created under.
-   * @throws Exception on errors while closing the client
+   * @param projectId the GCP project the Note belongs to
    */
-  public static void updateNote(Note updated, String noteId, String projectId) throws Exception {
-    try (ContainerAnalysisClient client = ContainerAnalysisClient.create()) {
-      final String noteName = client.formatNoteName(projectId, noteId);
-
-      client.updateNote(noteName, updated);
-    }
+  public static void updateNote(GrafeasV1Beta1Client client, Note updated, String noteId, 
+      String projectId) {
+    final String noteName = client.formatNoteName(projectId, noteId);
+    UpdateNoteRequest request = UpdateNoteRequest.newBuilder()
+                                                 .setName(noteName)
+                                                 .setNote(updated)
+                                                 .build();
+    client.updateNote(request);
   }
   // [END update_note]
 
   // [START update_occurrence]
   /**
-   * Makes an update to an existing occurrence
-   *
+   * Pushes an update to an Occurrence that already exists on the server
+   * @param client The Grafeas client used to perform the API requests.
    * @param occurrenceName the name of the occurrence to delete.
    *                       format: "projects/{projectId}/occurrences/{occurrence_id}"
```

Comment:
All the forms should be stylized as:
"projects/[PROJECT_ID]/occurrences/[OCCURRENCE_ID]"